### DISCUSSION TITLE
Explicit commit message renderer

### DISCUSSION THREAD

**Discussion by aalmiray** [jreleaser/jreleaser#826] [Ideas]
As of `v1.0.0` one may configure how a commit message may be rendered in the changelog by using mustache templates matching commit values such as title, body, author. However, these values will be rendered as is. There are two open tickets where the ability to render these values differently is desired:

#806 
#809 

Perhaps a collection of custom renderers may be required. A renderer may react to labels attached to a commit, use regex to transform title, body, or both at the same time. The trick now is how should these transformations be defined?

cc @gotson

**Comment by gotson** [comment]
Hi @aalmiray , i'm sorry i didn't came back to you earlier.

I'm not sure what custom renderers would be. Would that be some custom classes, or some configuration ?

**Reply by aalmiray** [reply]
Yes, that would be an additional configuration section in `changelog`. One can say there is a default renderer already whose overall formatting capabilities are found in expanded properties instead of at a given location. 

In a way, the `renderer` section should interact with the existing labeler/category pair. I have yet to find an intuitive way to express in the DSL how renderer(s) may be defined.
